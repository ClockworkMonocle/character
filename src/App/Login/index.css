.login__title {
  max-width: 220px;
  margin: 0 auto;
  text-align: center;
  padding-top: 24px;
}
