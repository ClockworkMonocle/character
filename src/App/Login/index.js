import React, { Component } from 'react';
import PropTypes from 'prop-types';
import firebaseui from 'firebaseui';
import firebase from 'firebase';
import Type from 'components/Type';
import Icon from 'components/Icon';

import 'firebaseui/dist/firebaseui.css';
import './index.css';

class Login extends Component {
  static propTypes = {
    history: PropTypes.shape({
      push: PropTypes.func.isRequired,
    }).isRequired,
  };

  componentDidMount() {
    const config = {
      signInOptions: [
        {
          provider: firebase.auth.GoogleAuthProvider.PROVIDER_ID,
          scopes: [
            'https://www.googleapis.com/auth/userinfo.profile',
            'https://www.googleapis.com/auth/userinfo.email',
          ],
        },
        // firebase.auth.FacebookAuthProvider.PROVIDER_ID,
        // firebase.auth.TwitterAuthProvider.PROVIDER_ID,
        // firebase.auth.GithubAuthProvider.PROVIDER_ID,
        firebase.auth.EmailAuthProvider.PROVIDER_ID,
        // firebase.auth.PhoneAuthProvider.PROVIDER_ID,
      ],
      callbacks: {
        signInSuccess: /* istanbul ignore next */ () =>
          this.props.history.push('/app'),
      },
    };

    let fbui = firebaseui.auth.AuthUI.getInstance();
    /* istanbul ignore next */
    if (fbui) {
      /* istanbul ignore next */
      fbui.reset();
    } else {
      fbui = new firebaseui.auth.AuthUI(firebase.auth());
    }

    fbui.start('#login', config);
  }

  render() {
    return [
      <div key="login__title" className="login__title">
        <Icon name="user-secret" size={64} />
        <Type variant="display-1">Who are you?</Type>
      </div>,
      <div key="login__form" id="login" />,
    ];
  }
}

export default Login;
