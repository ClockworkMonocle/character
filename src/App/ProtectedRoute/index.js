import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Route, Redirect } from 'react-router-dom';

export function ProtectedRoute({ user, ...routeProps }) {
  return (
    <Route
      path={routeProps.path}
      render={() => {
        if (!user.profile) {
          return <Redirect to="/login" />;
        }

        return <Route {...routeProps} />;
      }}
    />
  );
}

ProtectedRoute.propTypes = {
  user: PropTypes.shape({
    profile: PropTypes.object,
  }).isRequired,
};

const mapStateToProps = state => ({
  user: state.user,
});

export default connect(mapStateToProps)(ProtectedRoute);
