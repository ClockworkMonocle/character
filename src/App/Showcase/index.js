import React from 'react';
import DocumentTitle from 'react-document-title';

export default function Showcase() {
  return (<DocumentTitle title="Showcase">
    <section>
      <h1>Heading 1</h1>
      <h2>Heading 2</h2>
      <h3>Heading 3</h3>
      <p>Body copy</p>
      <p className="fa fa-user" />
      <p className="icon-shield" />
    </section>
  </DocumentTitle>);
}
