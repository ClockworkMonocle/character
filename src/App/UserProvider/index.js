import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import LoadingIndicator from 'components/LoadingIndicator';
import Type from 'components/Type';

import { authenticateUser } from 'state/user/actions';

import './index.css';

class UserProvider extends Component {
  static propTypes = {
    authenticateUser: PropTypes.func.isRequired,
    user: PropTypes.shape({
      isLoading: PropTypes.bool,
      error: PropTypes.string,
    }),
  };

  componentWillMount() {
    this.props.authenticateUser();
  }

  render() {
    const { user, children } = this.props;

    if (!user || user.isLoading) {
      return (
        <section className="user_provider__container">
          <LoadingIndicator />
        </section>
      );
    }

    if (user.error) {
      return (
        <section className="user_provider__container">
          <Type element="h1" variant="display-1">
            Some error occurred
          </Type>
          <Type element="p" variant="body">
            <Link to="/login">Try logging in again</Link>
          </Type>
        </section>
      );
    }

    return children;
  }
}

const mapStateToProps = /* istanbul ignore next */ state => ({
  user: state.user,
});

const mapDispatchToProps = {
  authenticateUser,
};

export { UserProvider };
export default connect(mapStateToProps, mapDispatchToProps)(UserProvider);
