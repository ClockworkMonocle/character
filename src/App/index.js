import React from 'react';
import DocumentTitle from 'react-document-title';
import { Provider } from 'react-redux';
import { BrowserRouter, Switch, Redirect, Route } from 'react-router-dom';

import Login from './Login';

export function App({ store }) {
  return (
    <DocumentTitle title="D&D 5e Character">
      <Provider store={store}>
        <BrowserRouter>
          <Switch>
            <Route path="/login" exact component={Login} />
            <Route
              path="/profile"
              exact
              render={/* istanbul ignore next */ () => (
                <h1>This one is profile</h1>
              )}
            />
            <Route
              path="/app"
              render={/* istanbul ignore next */ () => <h1>This one is app</h1>}
            />
            <Route
              render={/* istanbul ignore next */ () => <Redirect to="/app" />}
            />
          </Switch>
        </BrowserRouter>
      </Provider>
    </DocumentTitle>
  );
}

App.propTypes = {
  store: Provider.propTypes.store,
};

export default App;
