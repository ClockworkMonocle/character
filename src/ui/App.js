'use strict';

import React from 'react';

import Landing from './Landing/Landing';
import Character from './Character/Character';
import Login from './Login/Login';
import Profile from './Profile/Profile';
import HTML404 from './NotFound/HTML404';
import { ROUTE_LANDING, ROUTE_LOGIN, ROUTE_PROFILE, ROUTE_CHARACTER } from './routes';

export default React.createClass({
  display: 'App',

  render() {
    let route = this.props.state.route.get('route');
    let { dispatch, state } = this.props;

    switch (route) {
      case ROUTE_LANDING:
        return <Landing dispatch={dispatch} state={state}/>;
      case ROUTE_LOGIN:
        return <Login dispatch={dispatch} state={state}/>;
      case ROUTE_PROFILE:
        return <Profile dispatch={dispatch} state={state} />;
      case ROUTE_CHARACTER:
        return <Character dispatch={dispatch} state={state}/>;
      default:
        return <HTML404 dispatch={dispatch} state={state}/>;
    }
  },
});
