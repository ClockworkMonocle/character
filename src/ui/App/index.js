import React, { Component } from 'react';
import { Switch, Route, Redirect } from 'react-router-dom';

import ProtectedRoute from '../ProtectedRoute';
import UserProvider from '../UserProvider';
import Profile from '../Profile/Profile';
import Character from '../Character/Character';

export default class App extends Component {
  render() {
    const { state, dispatch, match } = this.props;
    return (<UserProvider state={state} dispatch={dispatch}>
      <Switch>
        <ProtectedRoute
          state={state}
          exact
          path={`${match.url}/profile`}
          render={() => <Profile state={state} dispatch={dispatch} />}
        />

        <ProtectedRoute
          state={state}
          exact
          path={`${match.url}/character/:characterId`}
          render={matchProps => <Character
            state={state}
            dispatch={dispatch}
            characterId={matchProps.match.params.characterId}
          />}
        />

        <Route render={() => <Redirect to={`${match.url}/profile`} />} />
      </Switch>
    </UserProvider>);
  }
}
