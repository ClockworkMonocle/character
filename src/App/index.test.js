import React from 'react';
import { shallow } from 'enzyme';
import App from './index';

test('<App /> renders', () => {
  const mockStore = {
    getState() {},
    subscribe() {},
    dispatch() {},
  };

  expect(shallow(<App store={mockStore} />)).toMatchSnapshot();
});
