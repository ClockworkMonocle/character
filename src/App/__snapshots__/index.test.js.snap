// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders 1`] = `
<SideEffect(DocumentTitle)
  title="D&D 5e Character"
>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "subscribe": [Function],
      }
    }
  >
    <BrowserRouter>
      <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/login"
        />
        <Connect(UserProvider)>
          <Switch>
            <Connect(ProtectedRoute)
              path="/app"
              render={[Function]}
            />
            <Route
              render={[Function]}
            />
          </Switch>
        </Connect(UserProvider)>
      </Switch>
    </BrowserRouter>
  </Provider>
</SideEffect(DocumentTitle)>
`;
